### DISCUSSION TITLE
Nextcloud inaccessible when redis is down

### DISCUSSION THREAD

**Discussion by 2fst4u** [nextcloud/helm#501] [Q&A]
I have redis as my distributed memcache and it works fine, but I'm trying to change how it works. I've stumbled upon the docs saying you ideally shouldn't use redis for local memcache and instead use apcu. I've also found the following website:

https://www.root101.net/nextcloud-highly-available-cluster-notes/

Which states "Using APCu for local cache is a better idea, talking with an external server slows down the up. Distributed cache needs to be the same for all cluster nodes. So external REDIS server is used for this cache. When REDIS stops, cluster slows down but keeps going still."

Which is broken english, but I gather they mean this will allow nextcloud to continue working, albeit slower, when redis is down. Is this correct? Because when I stop redis it causes my nextcloud instance to have an internal server error until redis comes back up.

**Comment by provokateurin** [comment]
That documentation sounds wrong. If you have redis configured and the redis server is down you always get an internal server error as you described it. Please check to the official documentation and not random websites

**Reply by 2fst4u** [reply]
As per official documentation:

```
Furthermore, you could use Redis for the local cache like so, but it’s not recommended (see warning below):

'memcache.local' => '\OC\Memcache\Redis',

Warning

Using Redis for local cache on a multi-server setup can cause issues. Also, even on a single-server setup, APCu (see section above) should be faster.
```

Where it is also recommended to use the configuration quoted. Why use apcu for local memcache if redis is required to always be up and running?

**Reply by provokateurin** [reply]
No you don't need always need redis, only if you configure it.

**Reply by 2fst4u** [reply]
That's not my question. **If** redis is configured, which it is, why would a local memcache be configured if the redis configuration is always required to be running?

**Reply by provokateurin** [reply]
As far as I know there are different types of cache that are used. Redis will store user sessions and database caches while something like APCu helps the JIT compiler. I think you will find better answers online, also because this is not a helm or docker specific question.

**Reply by 2fst4u** [reply]
I haven't found any answers online after hours of troubleshooting, which is why I'm asking here.
